OpenEdXMobile/src/main/java/org/edx/mobile/util/JavaUtil.java

```diff
@@ -82,7 +82,7 @@ public static Double formatDoubleValue(@NonNull Double value, int places) {
      */
     @NonNull
     public static String truncateString(@NonNull String string, int maxChars) {
-        if (string.length() > maxChars) {
+        if (string != null && string.length() > maxChars) {
```

Comment:
I thought about something similar when reviewing this. We are not forcing failures on  NonNull annotations so I figured it would be okay to leave the defensive code in there, though based on the use case you just described, perhaps it's better to let this fail fast.